src/playlist.rs

```diff
@@ -977,8 +841,99 @@ mod test {
                 fourth_song.to_owned(),
                 second_song.to_owned()
             ],
-            closest_album_to_group_by_key(group, pool.to_owned(), |s| s.bliss_song.to_owned())
-                .unwrap(),
+            closest_album_to_group(group, pool.to_owned()).unwrap(),
         );
     }
+
+    // This test case is non-deterministic and could fail in rare cases.
+    #[test]
+    fn test_forest_options() {
+        // These songs contains analysis of actual music. Tracks 1-3 are recording of Mozart's
+        // piano concerto no. 19, 9-11 are a recording of Mozart's piano concerto no. 23, and
+        // tracks 4-8 are Miles Davis' "Kind Of Blue".
+        let first_song = Song {
```

Comment:
nit, but I think maybe naming them `mozart_training_1`, `mozart_training_2`, `miles_davis_1`, etc would be a bit more self-descriptive (especially useful when they're used towards the end)

Reply:
Agreed, I named them.